File: 23.py
# -*- coding:utf-8 -*-
import copy
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def FindKthToTail(self, head:ListNode, k:int)->ListNode:
        if(head==None):
            return None
        pNode = copy.deepcopy(head)
        pKNode = copy.deepcopy(head)
        for _ in range(1,k-1):
            pNode = pNode.next
            if(pNode==None):
                return None
        while(pNode.next!=None):
            pNode = pNode.next
            pKNode = pKNode.next
        return pKNode
# write code here

File: 23.cpp
//
// Created by zb198 on 2019/10/3.
//

struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(nullptr) {
	}
};
class Solution {
public:
    ListNode* FindKthToTail(ListNode* pListHead, unsigned int k) {
        if(pListHead== nullptr){
            return nullptr;
        }
        ListNode* pKNode = pListHead;
        for (int i = 1; i <=k-1 ; ++i) {
            pListHead = pListHead->next;
            if(pListHead== nullptr){
                return nullptr;
            }
        }
        while (pListHead->next!= nullptr){
            pListHead = pListHead->next;
            pKNode = pKNode->next;
        }
        return pKNode;

    }
};
